src/openzaak/components/documenten/api/filters.py

```diff
@@ -37,3 +43,40 @@ class Meta:
             "startdatum": ["lt", "lte", "gt", "gte"],
             "einddatum": ["lt", "lte", "gt", "gte"],
         }
+
+
+def object_queryset(request):
+    object_value = request.query_params.get("object", "")
+    if "zaken" in object_value:
+        return Zaak.objects.all()
+    return Besluit.objects.all()
+
+
+class ObjectFilter(URLModelChoiceFilter):
+    def filter(self, qs, value):
+        if isinstance(value, Zaak):
+            self.field_name = "zaak"
+        else:
```

Comment:
I'd make an explicit check here for `Besluit` instance and if it's none of both, raise a `TypeError`. Now this will silently do weird stuff if a third model gets added at some point.

Reply:
fixed